tests/Resources/Page/LoginPage.robot

```diff
@@ -34,7 +34,9 @@ Select Login Authentication Type
 Login To Openshift
    [Arguments]  ${ocp_user_name}  ${ocp_user_pw}  ${ocp_user_auth_type}
     # Give the login prompt time to render after browser opens
-    Wait Until Element is Visible  xpath://div[@class="pf-c-login"]  timeout=15seconds
+    # FIXME: this wastes $timeout seconds every time it's called if 
+    # no login screen appears
+    Wait Until Element is Visible  xpath://div[@class="pf-c-login"]  timeout=10s
```

Comment:
I've investigated a little bit about this and I think there is another approach that could work to optimize this keyword:
- if the url starts with https://oauth-openshift  login is required and we need to wait for the elements to be visible
- if the url is something else (e.g. https://jupyter-nb-ldap-2dadmin13-rhods-notebooks.apps.qeaisrhods-mon.XXXX.s1.devshift.org/notebook/rhods-notebooks/jupyter-nb-ldap-2dadmin13/lab) it means that there was a session already opened and we can just return from this keyword without waiting

Reply:
This should work. I've taken a first stab at it but we probably need to test is extensively before merging

Reply:
Thanks!